### PR TITLE
fix: errors and config validation

### LINKED ISSUE #4539: issue with -c api.port
Having a space between `api.port` and the value causes the port to be read as `1`

```
→ bacalhau serve --orchestrator  -c orchestrator.port 2222 -c api.port=3333
16:34:09.008 | INF ../ProtocolLabs/workspace/bacalhau/cmd/cli/serve/serve.go:102 > Config loaded from: [], and with data-dir /var/folders/vz/pk7pk9qx1c5gbktg8nhyl88h0000gn/T/bacalhau_test.AjhLLURDzE
16:34:09.018 | INF ../ProtocolLabs/workspace/bacalhau/cmd/cli/serve/serve.go:228 > Starting bacalhau...
16:34:09.054 | INF ../ProtocolLabs/workspace/bacalhau/cmd/cli/serve/serve.go:299 > bacalhau node running [address:0.0.0.0:3333] [compute_enabled:false] [name:n-ed0ddc0f-40af-4ce5-8c24-6d6d39649321] [orchestrator_address:0.0.0.0:1] [orchestrator_enabled:true] [webui_enabled:false]

To connect to this node from the local client, run the following commands in your shell:
export BACALHAU_API_HOST=127.0.0.1
export BACALHAU_API_PORT=3333
```

```
→ bacalhau serve --orchestrator  -c orchestrator.port 2222 -c api.port 3333
16:34:21.781 | INF ../ProtocolLabs/workspace/bacalhau/cmd/cli/serve/serve.go:102 > Config loaded from: [], and with data-dir /var/folders/vz/pk7pk9qx1c5gbktg8nhyl88h0000gn/T/bacalhau_test.AjhLLURDzE
16:34:21.799 | INF ../ProtocolLabs/workspace/bacalhau/cmd/cli/serve/serve.go:228 > Starting bacalhau...

Error: address http://0.0.0.0:1 is already in use
Hint:  To resolve this, either:
1. Check if you are already running bacalhau
2. Stop the other process using the port
3. Configure a different port using one of these methods:
   a. Use the `-c api.port=<new_port>` flag with your serve command
   b. Set the port in a configuration file with `bacalhau config set api.port=<new_port>`

### LINKED ISSUE #4546: config set is not consistent with other commands
`config set` accepts space separated keys and values, and this behavour is printed first in the `--help` output, but this format is not compatible in `bacalhau serve` and all other commands that accept key/value pair of configurations. 

Either provide support for space separated values everywhere, or remove it from `config set` as well. If we decide to remove it, then we need to provide a clear error message to the user

```
→ bacalhau config set --help
The 'set' command allows you to modify configuration values in your Bacalhau configuration file.

This command supports two input formats for setting configuration values:

1. Key-Value Pair Format:
   You can provide the key and value as separate arguments.
   Example:
     bacalhau config set api.host 127.0.0.1
     bacalhau config set compute.orchestrators http://127.0.0.1:1234 http://1.1.1.1:1234
     bacalhau config set compute.labels foo=bar,baz=buz

2. Key=Value Format:
   Alternatively, you can pass the key and value together in a single argument using the 'key=value' format.
   Example:
     bacalhau config set api.host=127.0.0.1
     bacalhau config set compute.orchestrators=http://127.0.0.1:1234,http://1.1.1.1:1234
     bacalhau config set compute.labels=foo=bar,baz=buz

Usage:
  /Users/walid/bacalhau-versions/v1.5.0/bacalhau config set [flags]
```

### LINKED ISSUE #4547: --api-host has lower priority than -c api.host
bacalhau serve --orchestrator -c config.yaml -c api.host=1.1.1.1 --api-host 3.3.3.3
resolves into 1.1.1.1 instead of the dedicated flag 3.3.3.3

### PR DESCRIPTION
**Pull request by frrist**
- closes #4539
```shell
$ bacalhau serve --orchestrator  -c orchestrator.port 2222 -c api.port=3333

Error: invalid argument "orchestrator.port" for "-c, --config" flag: config flag key "orchestrator.port" requires a value
```
- closes #4547
```shell
$ bacalhau serve --orchestrator -c config.yaml -c api.host=1.1.1.1 --api-host 3.3.3.3

Error: flag: --api-host and config flag key "api.host" cannot both be provided. Only one may be used
Hint:  Remove --api-host or --confg/-c api.host=1.1.1.1 from the command
```
- closes #4546
```shell
$ bacalhau config set --help
The 'set' command allows you to modify configuration values in your Bacalhau configuration file.

This command supports two input formats for setting configuration values:

 Key=Value Format:
   Alternatively, you can pass the key and value together in a single argument using the 'key=value' format.
   Example:
     bacalhau config set api.host=127.0.0.1
     bacalhau config set compute.orchestrators=http://127.0.0.1:1234,http://1.1.1.1:1234
     bacalhau config set compute.labels=foo=bar,baz=buz

Usage:
  ./bin/linux/amd64/bacalhau config set [flags]

Flags:
  -c, --config string   Path to the config file (default is $BACALHAU_DIR/config.yaml)
  -h, --help            help for set

Global Flags:
      --api-host string   The host for the client and server to communicate on (via REST).
                          Ignored if BACALHAU_API_HOST environment variable is set. (default "0.0.0.0")
      --api-port int      The port for the client and server to communicate on (via REST).
                          Ignored if BACALHAU_API_PORT environment variable is set. (default 1234)
      --data-dir string   The filesystem path bacalhau inits or opens a repo in (default "/home/frrist/.bacalhau")
```

### CODE REVIEW

**Review by udsamani — APPROVED** [review]

File: cmd/cli/config/set.go [hunk 1]
```diff
@@ -26,14 +26,7 @@ func newSetCmd() *cobra.Command {
...
 
-1. Key-Value Pair Format:
-   You can provide the key and value as separate arguments.
-   Example:
-     bacalhau config set api.host 127.0.0.1
-     bacalhau config set compute.orchestrators http://127.0.0.1:1234 http://1.1.1.1:1234
-     bacalhau config set compute.labels foo=bar,baz=buz
-
-2. Key=Value Format:
+ Key=Value Format:
```
Comment (wdbaruni) [inline cmd/cli/config/set.go]: The help message is still indicating two input formats `This command supports two input formats` and the use of the term `Alternatively`

File: cmd/util/flags/cliflags/config.go [hunk 2]
```diff
@@ -95,7 +97,20 @@ func (cf *ConfigFlag) parseReadMode() error {
...
 		// Handle dot separated path with boolean value
-		return setIfValid(viper.GetViper(), cf.Value, true)
+		key := strings.ToLower(cf.Value)
+		typ, ok := types.AllKeys()[key]
+		if !ok {
+			if _, err := os.Stat(key); err == nil {
+				return fmt.Errorf("config files must end in suffix '.yaml' or '.yml'")
+			}
+			return fmt.Errorf("no config key matching %q run 'bacalhau config list' for a list of valid keys", key)
+		}
```
Comment (wdbaruni) [inline cmd/util/flags/cliflags/config.go]: this validation and error message is duplicated in `setIfValid`

File: pkg/config/config.go [hunk 3]
```diff
@@ -141,6 +142,21 @@ func New(opts ...Option) (*Config, error) {
...
+	for name, flags := range c.flags {
+		for _, flag := range flags {
+			for key, value := range c.values {
+				if flag.Changed {
+					if name == key {
+						return nil, bacerrors.New("flag: --%s and config flag key %q cannot both be provided. Only one may be used", flag.Name, key).
+							WithHint("Remove --%s or --confg/-c %s=%s from the command", flag.Name, key, value)
+					}
+				}
+			}
```
Comment (wdbaruni) [inline pkg/config/config.go]: a bit confusing and difficult to read through. Can use better var names. Also I am sure we don't need three for loops here

Comment (wdbaruni) [reply pkg/config/config.go]: Also please add unit tests

**Pull request merged**